05-JWT-Basics/preferred/controllers/main.js

```diff
@@ -0,0 +1,21 @@
+const jwt = require('jsonwebtoken');
+
+const logon = async (req, res) => {
+  const { name, password } = req.body;
+  if (!name.trim() || !password.trim()) {
+    return res.status(400).json({ message: 'Name and password are required' });
+  }
+  const token = jwt.sign({ name }, process.env.JWT_SECRET, {
+    expiresIn: '24h',
+  });
+  res.status(200).json({ msg: 'user created', token });
+};
+
+const hello = async (req, res) => {
+  let { name } = req.user;
+  res.status(200).json({
+    msg: `Hello, ${name}!`,
+  });
```

Comment:
If the name comes from the users request is it possible for the user to lie about their name?

Reply:
Yes, if the name comes from the user's request and we don't have any validation set up it's possible for the user to lie about their name.

Reply:
Does the name come from the users request in this case?

Reply:
The name value is coming from a verified and authenticated JWT included in the user's request. There is a reason why we store the name value in the ```req.user``` object - this is a standard way to pass data between middleware functions and route handlers.